x/schema/keeper/msg_server_schema.go

```diff
@@ -69,6 +69,39 @@ func (k msgServer) CreateSchema(goCtx context.Context, msg *types.MsgCreateSchem
 }
 
 func (k msgServer) DeprecateSchema(goCtx context.Context, msg *types.MsgDeprecateSchema) (*types.MsgDeprecateSchemaResponse, error) {
-	// TODO: implement
-	return nil, errors.New("unimplemented")
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	err := msg.ValidateBasic()
+	if err != nil {
+		return nil, err
+	}
+
+	schemas, foundSchemas := k.GetWhatIsFromCreator(ctx, msg.GetCreator())
+	if !foundSchemas {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "No Schemas found under same creator as message creator.")
+	}
+
+	var schemaWI types.WhatIs
+	var foundSchemaWI bool
+	for _, a := range schemas {
+		if a.GetDid() == msg.GetDid() {
+			schemaWI = a
+			foundSchemaWI = true
+		}
+	}
+
+	if !foundSchemaWI {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "No Schema with same creator as message creator found.")
+	}
+
+	//If already deactivated, do nothing.
+	//Responsibility of caller to check if isActive beforehand
+	if schemaWI.GetIsActive() {
+		schemaWI.IsActive = false
+		k.SetWhatIs(ctx, schemaWI)
+	}
+
+	return &types.MsgDeprecateSchemaResponse{
+		Code:    http.StatusOK,
```

Comment:
we should change this to be a constant as to remove the http/net dep from this pkg. we can just use the value '200' for now

Reply:
Change made